src/LayerEditor/ui/data/regions.py

```diff
@@ -0,0 +1,75 @@
+from LayerEditor.ui.data.region import Region
+
+
+class Regions:
+    def __init__(self, diagram):
+        self.regions = {Region.none.id: Region.none}
+        self._diagram = diagram
+
+    def add_region(self, color=None, name="", dim=0):
+        reg = Region(id=None, color=color, name=name, dim=dim)
+        self.regions[reg.id] = reg
+        return reg.id
+
+    def delete_region(self, id):
+        del self.regions[id]
+
+    def get_region_names(self):
+        return [reg.name for reg in self.regions.values()]
+
+    def get_shape_region(self, shape_key):
+        dim, shape_id = shape_key
+        attr = None
+        if dim == 1:
```

Comment:
Can use `decomposition.shapes[dim][shape_id].attr`.

Reply:
Used in branch TB_LE_inicialization_and_IO